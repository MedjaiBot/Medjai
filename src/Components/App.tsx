import * as React from 'react';

export class App extends React.Component<{}, {}> {
    render () {
        return <div>
            <p>Hello World!</p>
        </div>;
    }
}
