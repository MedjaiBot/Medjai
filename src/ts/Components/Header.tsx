import * as React from 'react'
import { Menu, MenuItem } from 'semantic-ui-react'
import { Link } from 'react-router-dom'

export class Header extends React.Component<{}, {}>
{
    render ()
    {
        return <Menu inverted>
            <MenuItem>
                <Link to="/">Homepage</Link>
            </MenuItem>
            <MenuItem>
                <Link to="/extensions">Extensions</Link>
            </MenuItem>
            <MenuItem>
                <Link to="/settings">Settings</Link>
            </MenuItem>
        </Menu>
    }
}
