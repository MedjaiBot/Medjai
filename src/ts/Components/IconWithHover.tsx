import * as React from 'react'
import { Icon, IconProps } from 'semantic-ui-react'

export class IconWithHover extends React.Component<IconProps, {}>
{
    state = {
        showElement: false
    }

    render ()
    {
        return <span style={{
            paddingRight: '5px'
        }}>
            <Icon inverted name={this.props.name} onMouseEnter={() =>
            {
                this.setState({
                    showElement: true
                })
            }} onMouseLeave={() =>
            {
                this.setState({
                    showElement: false
                })
            }}/>
            {this.state.showElement ? this.props.children : null}
        </span>
    }
}
