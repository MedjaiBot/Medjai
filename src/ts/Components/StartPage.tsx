import * as React from 'react'
import { Container, Feed, Segment } from 'semantic-ui-react'
import { IconWithHover } from './IconWithHover'

export class StartPage extends React.Component<{}, {}>
{
    state = {
        currentElement: ''
    }

    render ()
    {
        return <div>
            <Container>
                <Segment inverted>
                    <Feed>
                        <Feed.Event>
                            <Feed.Label>
                                <img
                                    src='https://static-cdn.jtvnw.net/jtv_user_pictures/iseven-profile_image-8e0392e49ab2bf95-70x70.png'/>
                            </Feed.Label>
                            <Feed.Content>
                                <Feed.Summary>
                                    <Feed.User>iSEVEN</Feed.User>: Hello! <img
                                    src="https://static-cdn.jtvnw.net/emoticons/v1/25/1.0" alt="Kappa emote"
                                    className={'twitch-emote'}/>
                                    <Feed.Date>1 Hour Ago</Feed.Date>
                                </Feed.Summary>
                                <Feed.Meta>
                                    <Feed.Like>
                                        <IconWithHover name={'mail'}>
                                            <span>Send private Message</span>
                                        </IconWithHover>
                                        <IconWithHover name={'alarm mute'}>
                                            <span>Timeout</span>
                                        </IconWithHover>
                                        <IconWithHover name={'ban'}>
                                            <span>Ban</span>
                                        </IconWithHover>
                                    </Feed.Like>
                                </Feed.Meta>
                            </Feed.Content>
                        </Feed.Event>
                        <Feed.Event>
                            <Feed.Label>
                                <img
                                    src='https://static-cdn.jtvnw.net/jtv_user_pictures/iseven-profile_image-8e0392e49ab2bf95-70x70.png'/>
                            </Feed.Label>
                            <Feed.Content>
                                <Feed.Summary>
                                    <Feed.User>iSEVEN</Feed.User>: lul! <img
                                    src="https://static-cdn.jtvnw.net/emoticons/v1/25/1.0" alt="Kappa emote"
                                    className={'twitch-emote'}/>
                                    <Feed.Date>1 Hour Ago</Feed.Date>
                                </Feed.Summary>
                                <Feed.Meta>
                                    <Feed.Like>
                                        <IconWithHover name={'mail'}>
                                            <span>Send private Message</span>
                                        </IconWithHover>
                                        <IconWithHover name={'alarm mute'}>
                                            <span>Timeout</span>
                                        </IconWithHover>
                                        <IconWithHover name={'ban'}>
                                            <span>Ban</span>
                                        </IconWithHover>
                                    </Feed.Like>
                                </Feed.Meta>
                            </Feed.Content>
                        </Feed.Event>
                    </Feed>
                </Segment>
            </Container>
        </div>
    }
}
