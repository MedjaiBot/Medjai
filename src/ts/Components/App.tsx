import * as React from 'react';
import './../../scss/app.scss';
import {BrowserRouter as Router, Route, Switch} from 'react-router-dom'
import {Container} from "semantic-ui-react";

import {Header} from "./Header";
import {StartPage} from "./StartPage";

export class App extends React.Component<{}, {}> {
    render() {
        return <div>
            <Router>
                <Container fluid>
                    <Header/>
                    <Switch>
                        <Route path={"/"} exact render={() => {
                            return <StartPage/>;
                        }}>
                        </Route>
                        <Route path={"/extensions"} render={() => {
                            return <p>Extensions</p>;
                        }}/>
                        <Route path={"/settings"} render={() => {
                            return <p>Settings</p>;
                        }}/>
                    </Switch>
                </Container>
            </Router>
        </div>;
    }
}
