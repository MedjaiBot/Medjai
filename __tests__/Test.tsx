import * as React from 'react';
import renderer from 'react-test-renderer';
import {v4} from 'uuid';
import {App} from '../src/Components/App';
 
describe('General tests', () => {
    it('UUIDs should be 36 characters long', () => {
        const uuid = v4();

        expect(uuid.length).toBe(36);
    });
});

describe('App component tests', () => {
    it('should match the snapshot', () => {
        const tree = renderer.create(<App/>).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
