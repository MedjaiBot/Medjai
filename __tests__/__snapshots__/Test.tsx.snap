// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component tests should match the snapshot 1`] = `
<div>
  <p>
    Hello World!
  </p>
</div>
`;
